### PR TITLE
Bump minimum compile jdk version to 15

### PR DESCRIPTION
**Pull request by mark-vieira**
We're dropping support for Java 14 given that it's EOL so we should be building with Java 15 now.

### CODE REVIEW

**Comment by jasontedor** [comment]
@mark-vieira Have we communicated to the rest of the team that this requirement is coming? For example, it could break the performance team if they haven't already migrated to using JDK 15 for compilation.

**Comment by jasontedor** [comment]
I checked with the performance team and they are prepared for this change.

**Comment by mark-vieira** [comment]
I'll definitely send a shout out before merging the change. If for no other reason then folks will need to have Java 15 installed, etc. It's safe to bump JAVA_HOME, and then pull these changes to be uninterrupted.

**Review by jasontedor — APPROVED** [review]
LGTM.

File: buildSrc/src/integTest/java/org/elasticsearch/gradle/precommit/ThirdPartyAuditTaskIT.java [hunk 1]
```diff
@@ -49,18 +49,6 @@ public void testElasticsearchIgnored() {
         assertNoDeprecationWarning(result);
     }
 
-    public void testWithEmptyRules() {
```
Comment (pugnascotia) [inline buildSrc/src/integTest/java/org/elasticsearch/gradle/precommit/ThirdPartyAuditTaskIT.java:52]: Why has this been removed?

Comment (mark-vieira) [reply buildSrc/src/integTest/java/org/elasticsearch/gradle/precommit/ThirdPartyAuditTaskIT.java:52]: Forbidden api behavior has changed here. It no longer errors on empty rule files. Only if file itself is missing or not specified. I don't think it's worth adding additional checks on our end to reinstate that behavior.

**Comment by mark-vieira** [comment]
Ooops, merged this by mistake. I wanted to give folks a couple days to ensure they have a JDK 15 install. I've reverted and will push again later in the week.

**Pull request merged**